### PR TITLE
Assume aws role if provided

### PR DESCRIPTION
**Pull request by neowulf33**
* Remove all aws libraries except for sts and ecr
* Autodetect role name based on the provided profile

Referencing ticket https://github.com/sbilinski/sbt-ecr/issues/9

### CODE REVIEW

**Comment by sbilinski** [comment]
Thanks for the contribution @neowulf33. 

> Remove all aws libraries except for sts and ecr

This is a really cool improvement! Please consider opening a separate issue/PR for it as it will make the "changelog" more obvious (and the discussion simpler).   

> Autodetect role name based on the provided profile

I feel this is out of scope of this plugin. Let me tell you how it works in my current project:

We basically follow [this article](https://aws.amazon.com/blogs/security/how-to-implement-a-general-solution-for-federated-apicli-access-using-saml-2-0/) from Amazon. The idea is, that you authenticate once, save the credentials from the token and then run **all** your AWS commands without worrying about the role assumption anymore. 

Check out the [script](https://s3.amazonaws.com/awsiammedia/public/sample/SAMLAPICLIADFS/samlapi_formauth.py) from the article. It does exactly what you want, except that it saves the token credentials for later: 

    token = conn.assume_role_with_saml(role_arn, principal_arn, assertion)
    ...
    config.set('saml', 'aws_access_key_id', token.credentials.access_key)
    config.set('saml', 'aws_secret_access_key', token.credentials.secret_key)
    config.set('saml', 'aws_session_token', token.credentials.session_token)
    ...
    config.write(configfile)

Bottom line is: the role assumption should be handled by a different tool/plugin, which should be executed before you interact with AWS in any way.

**Comment by sbilinski** [comment]
I'm closing this PR for now, given the reasoning above. Please feel free to open another one for the library cleanup - I'd really appreciate it!

**Comment by neowulf33** [comment]
Thank you. I have added a comment to https://github.com/sbilinski/sbt-ecr/pull/9
